rfc/rfc-42/rfc-42.md

```diff
@@ -0,0 +1,220 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+# RFC-42: Implement Compaction/Clustering Service for Hudi
```

Comment:
@yuzhaojing Thanks for raising the feature request.  RFC-42 already exists.  Could you raise a separate PR to secure a new RFC number first in `rfc/README.md` like this [one](https://github.com/apache/hudi/pull/4303)?  The new RFC process is described [here](https://hudi.apache.org/contribute/rfc-process/).

Reply:
Ok, I will do this.